src/src_python/antares_xpansion/driver.py

```diff
@@ -262,29 +263,44 @@ def pre_antares(self):
-        ini_file[self.config.OPTIMIZATION][self.config.USE_XPRS] = "false"
-        ini_file.remove_option(self.config.OPTIMIZATION, self.config.USE_XPRS)
-        ini_file.remove_option(self.config.OPTIMIZATION, self.config.INBASIS)
-        ini_file.remove_option(self.config.OPTIMIZATION, self.config.OUTBASIS)
-        if self.is_accurate():
-            ini_file['general']['mode'] = 'expansion'
-            ini_file['other preferences']['unit-commitment-mode'] = 'accurate'
-            ini_file[self.config.OPTIMIZATION]['include-tc-minstablepower'] = 'true'
-            ini_file[self.config.OPTIMIZATION]['include-tc-min-ud-time'] = 'true'
-            ini_file[self.config.OPTIMIZATION]['include-dayahead'] = 'true'
-        else:
-            ini_file['general']['mode'] = 'Economy'
-            ini_file['other preferences']['unit-commitment-mode'] = 'fast'
-            ini_file[self.config.OPTIMIZATION]['include-tc-minstablepower'] = 'false'
-            ini_file[self.config.OPTIMIZATION]['include-tc-min-ud-time'] = 'false'
-            ini_file[self.config.OPTIMIZATION]['include-dayahead'] = 'false'
 
-        with open(self.general_data(), 'w') as out_file:
-            ini_file.write(out_file)
+        def _line_is_not_a_section_header(_line):
+            return len(_line.split('=')) == 2
+
+        with open(self.general_data(), 'r') as reader:
+            lines = reader.readlines()
+
+        with open(self.general_data(), 'w') as writer:
+            current_section = ""
+            for line in lines:
+                if _line_is_not_a_section_header(line):
+                    key = line.split('=')[0].strip()
+                    line = self._get_new_line(line, current_section, key)
+                else:
+                    current_section = line.strip()
+
+                if line:
+                    writer.write(line)
+
+    def _get_new_line(self, line, section, key):
+        changed_val = {('['+self.config.OPTIMIZATION+']', self.config.EXPORT_MPS): 'true',
+                       ('['+self.config.OPTIMIZATION+']', self.config.EXPORT_STRUCTURE): 'true',
+                       ('['+self.config.OPTIMIZATION+']', 'include-tc-minstablepower'): 'true' if self.is_accurate() else 'false',
+                       ('['+self.config.OPTIMIZATION+']', 'include-tc-min-ud-time'): 'true' if self.is_accurate() else 'false',
+                       ('['+self.config.OPTIMIZATION+']', 'include-dayahead'): 'true' if self.is_accurate() else 'false',
+                       ('['+self.config.OPTIMIZATION+']', self.config.USE_XPRS): None,
+                       ('['+self.config.OPTIMIZATION+']', self.config.INBASIS): None,
+                       ('['+self.config.OPTIMIZATION+']', self.config.OUTBASIS): None,
+                       ('[general]', 'mode'): 'expansion' if self.is_accurate() else 'Economy',
+                       ('[other preferences]', 'unit-commitment-mode'): 'accurate' if self.is_accurate() else 'fast'
+                       }
```

Comment:
Only define `changed_val` once, instead of every call of `_get_new_line` ?

This could be done in `XpansionDriver`'s `__init__`.